src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,43 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int RANGE_BOUND = 5;
+    public static final int DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private Figure figure;
```

Comment:
The `figure` variable should be local in the `getRandomFigure` and `getDefaultFigure` methods instead of being a class-level variable. This ensures that each method call operates on its own instance and avoids potential side effects from shared state.